### ISSUE TITLE
Is the C source code open?

### ISSUE THREAD

**Issue by yulf** [Kickflip/kickflip-android-sdk#49]
Hi, I found you have used ffmpeg to mix the audio and video, Is the C source code open? Where can I get it?

**Comment by chrisballinger** [comment]
ffmpeg.org?

**Comment by yulf** [comment]
The author must have modified some codes to satisfy the application.  This is the key.

**Comment by chrisballinger** [comment]
No, the FFmpeg sources are unmodified. The JNI bridge is open source.

**Comment by yulf** [comment]
Where can I download the JNI bridge codes?

**Comment by chrisballinger** [comment]
https://github.com/Kickflip/kickflip-android-sdk/blob/master/sdk/src/main/java/net/openwatch/ffmpegwrapper/FFmpegWrapper.java

Cant seem to find the .c file, but it's somewhere.

**Comment by yulf** [comment]
Did the author release the C source code ? I can not find them.

**Comment by Arlem** [comment]
The FFmpeg wrapper can be found here: https://github.com/OpenWatch/FFmpegWrapper-Android/tree/master/FFmpegWrapper/jni

But, if you want to make changes to that code, you have to build the library with ndk-build.sh

EDIT: the compiled and built .so library goes here:
https://github.com/Kickflip/kickflip-android-sdk/tree/master/sdk/src/main/jniLibs

Pay attention to match library/architecture

**Comment by yulf** [comment]
In https://github.com/Kickflip/kickflip-android-sdk/blob/master/sdk/src/main/java/net/openwatch/ffmpegwrapper/FFmpegWrapper.java, there are  functions as follow:
  public native void setAVOptions(AVOptions jOpts);
    public native void prepareAVFormatContext(String jOutputPath);
    public native void writeAVPacketFromEncodedData(ByteBuffer jData, int jIsVideo, int jOffset, int jSize, int jFlags, long jPts);
    public native void finalizeAVFormatContext();

But,  In https://github.com/OpenWatch/FFmpegWrapper-Android/blob/master/FFmpegWrapper/jni/FFmpegWrapper.c, I can not find corresponding C implementations.

**Comment by Arlem** [comment]
prepareAVFormatContext -> https://github.com/OpenWatch/FFmpegWrapper-Android/blob/master/FFmpegWrapper/jni/FFmpegWrapper.c#L302
setAVOptions -> https://github.com/OpenWatch/FFmpegWrapper-Android/blob/master/FFmpegWrapper/jni/FFmpegWrapper.c#L345
writeAVPacketFromEncodedData -> https://github.com/OpenWatch/FFmpegWrapper-Android/blob/master/FFmpegWrapper/jni/FFmpegWrapper.c#L374
finalizeAVFormatContext -> https://github.com/OpenWatch/FFmpegWrapper-Android/blob/master/FFmpegWrapper/jni/FFmpegWrapper.c#L432
